thunder/tests/test_nvmath_executor.py

```diff
@@ -0,0 +1,40 @@
+from torch.testing import assert_close
+
+from thunder.core import dtypes, prims
+from thunder.core.devices import DeviceType
+from thunder.executors.nvmathex import nvmath_matmul_ex
+from thunder.tests.framework import ops, TestExecutor
+from thunder.tests.opinfos import opinfos
+
+
+class nvMathTestExecutor(TestExecutor):
+    name = "nvmath"
+    supported_devicetypes = (DeviceType.CUDA,)
+    supported_dtypes = (*dtypes.all_dtypes,)
+
+    def executors_list(self):
+        return [nvmath_matmul_ex]
+
+
+@ops((op for op in opinfos if op.name in ("matmul", "linear")), supported_executors=(nvMathTestExecutor(),))
+def test(op, device, dtype, executor, comparator):
+    for sample in op.sample_inputs(device, dtype, requires_grad=False):
+        # prims do not support broadcasting
+        if sample.args[0].shape[:-2] != sample.args[1].shape[:-2]:
```

Comment:
These are detailed manipulations of the sample inputs, and they make complete sense

One interesting extension to the OpInfos that might be interesting (and certainly is not for this PR) is to easily bind the sample inputs to the signatures of functions, so then instead of having to write

```
if len(sample.args) == 3:
```

or

```
if op.name == "linear" and len(sample.args) == 2:
```

a developer might be able to write something like

```
ba = sample.bind(op)
if ba['bias'] is not None:
  ...
```

Which isn't a huge improvement but it's a little more readable.